src/setup.jl

```diff
@@ -485,371 +583,324 @@ function System(coord_file::AbstractString,
                 data=nothing,
                 implicit_solvent=:none,
                 kappa=0.0u"nm^-1",
-                rename_terminal_res::Bool=true,
-                grad_safe::Bool=false) where AT <: AbstractArray
-    if dist_buffer < zero(dist_buffer)
-        throw(ArgumentError("dist_buffer ($dist_buffer) should not be less than zero"))
-    end
+                grad_safe::Bool=false) where {AT<:AbstractArray}
+
+    dist_buffer < zero(dist_buffer) && throw(ArgumentError("dist_buffer ($dist_buffer) should not be less than zero"))
     dist_neighbors = dist_cutoff + dist_buffer
     T = typeof(force_field.weight_14_coulomb)
 
-    # Chemfiles uses zero-based indexing, be careful
-    trajectory = Chemfiles.Trajectory(coord_file)
-    frame = Chemfiles.read(trajectory)
-    top = Chemfiles.Topology(frame)
+    resname_replacements  = force_field.residue_name_replacements
+    atomname_replacements = force_field.atom_name_replacements
+    standard_bonds        = force_field.standard_bonds
+
+    #return resname_replacements, atomname_replacements
+
+    # ---- Read structure ----
+    traj  = Chemfiles.Trajectory(coord_file)
+    frame = Chemfiles.read(traj)
+    top   = Chemfiles.Topology(frame)
     n_atoms = size(top)
 
-    if isnothing(boundary)
-        boundary_used = boundary_from_chemfiles(Chemfiles.UnitCell(frame), T,
-                                                (units ? u"nm" : NoUnits))
+    # ---- Boundary ----
+    boundary_used = isnothing(boundary) ? boundary_from_chemfiles(Chemfiles.UnitCell(frame), T, (units ? u"nm" : NoUnits)) : boundary
+    minimum(box_sides(boundary_used)) < (2 * dist_cutoff) && @warn "Minimum box side is less than 2 * dist_cutoff; this can be unphysical"
+
+    # ---- Units and coordinates ----
+    if units
+        coords = [T.(SVector{3}(col)u"nm" / 10.0) for col in eachcol(Chemfiles.positions(frame))]
     else
-        boundary_used = boundary
+        coords = [T.(SVector{3}(col) / 10.0) for col in eachcol(Chemfiles.positions(frame))]
     end
-    min_box_side = minimum(box_sides(boundary_used))
-    if min_box_side < (2 * dist_cutoff)
-        @warn "Minimum box side ($min_box_side) is less than 2 * dist_cutoff " *
-              "($(2 * dist_cutoff)), this can lead to unphysical simulations"
+    if center_coords
+        coords = coords .- (mean(coords),) .+ (box_center(boundary_used),)
     end
+    
+    canonical_system = canonicalize_system(top, resname_replacements, atomname_replacements)
+
+    top_bonds = create_bonds(canonical_system, standard_bonds)
+    top_bonds = create_disulfide_bonds(coords, boundary_used, canonical_system, top_bonds)
+    top_bonds = read_connect_bonds(coord_file, top_bonds, canonical_system) 
```

Comment:
Reading the CONECT records works for PDB files but not for other formats. Is it possible to extract extra bonds from the Chemfiles topology (which should include the CONECT bonds)?